internal/git/gitsync_processor_test.go

```diff
@@ -110,3 +134,320 @@ func Test_watchRepo(t *testing.T) {
 		})
 	}
 }
+
+/*
+	func updateFileInBranch(repo *git.Repository, branchName, fileName, content string, author *object.Signature, tagName string) error {
```

Comment:
Why are these tests are commented out?

Reply:
These tests are e2e tests ,which i was running with my git credentials ,I will fix them once I dockerize the git part as discussed on the call

Reply:
e2e tests, or do you mean unit tests which use  an actual non-mock repo? e2e tests would probably live in a separate directory 
 
I remember, the issue was that you needed to be able to write to the repo in the test (which I think could be part of a unit test)

Reply:
yeah i have to mock changes in the remote repo such that i can get the changes while cloning and checking for updates ,for that i was using my credentials through which i was pushing the changed file in github repo and then triggering the tests . I am thinking to remote git server in docker container which i can up and discard during tests and can connect with may be like http://localhost or ssh://

Reply:
Is it possible to follow the unit test in go git library which push to a git repository backed by a local storage? https://github.com/go-git/go-git/blob/master/repository_test.go#L1535

Reply:
this looks interesting ,let me try this out

Reply:
I fixed the test cases by creating local remote took reference from go git tests @xdevxy @juliev0